camel/societies/role_playing.py

```diff
@@ -270,16 +271,16 @@ def step(
             return (ChatAgentResponse([], False, {}),
                     ChatAgentResponse([], user_response.terminated,
                                       user_response.info))
-        user_msg = self.process_messages(user_response.msgs)
-        self.user_agent.update_messages(user_msg)
+        user_msg = self.make_critic_choice(user_response.msgs)
+        self.user_agent.submit_critic_choice(user_msg)
```

Comment:
Do not see the difference between `submit_critic_choice` and `update_messages`.

Reply:
An API must be as restrictive as you think is needed to make it protected from misuse. This is what I did. Anyway, this PR is the cleanup to make the role playing API fundamental and support multi-agent. I reckon we will rework it heavily soon.

Reply:
update messages is a very broad name given the method has project-specific logic. I would vote for "critic" in the name, yet somewhere the meaning of the word critic itself should be documented. Or we could come up with a new, even more self-descriptive name

Reply:
What about `reduce_choices()`? By analogy with reduce pattern?

Reply:
How about `make_critic_choice` -> `reduce_messages`, `submit_critic_choice` -> `submit_message` since sometime the critic does not make a choice, it can enter a message or edit a message?

Reply:
Maybe make_critic_choice -> reduce_options? Because reduce_messages sounds like you want to shrink each message.